File: lib/core/services/firebase_service.dart
import 'package:firebase_core/firebase_core.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:firebase_analytics/firebase_analytics.dart';
import 'package:firebase_crashlytics/firebase_crashlytics.dart';

class FirebaseService {
  static FirebaseAuth get auth => FirebaseAuth.instance;
  static FirebaseFirestore get firestore => FirebaseFirestore.instance;
  static FirebaseAnalytics get analytics => FirebaseAnalytics.instance;
  static FirebaseCrashlytics get crashlytics => FirebaseCrashlytics.instance;

  // Initialize crash reporting
  static Future<void> initializeCrashlytics() async {
    FlutterError.onError = (errorDetails) {
      crashlytics.recordFlutterFatalError(errorDetails);
    };
  }

  // Log analytics event
  static Future<void> logEvent(String name, Map<String, dynamic>? parameters) async {
    await analytics.logEvent(name: name, parameters: parameters);
  }

  // Set user properties
  static Future<void> setUserProperty(String name, String value) async {
    await analytics.setUserProperty(name: name, value: value);
  }
}

File: src/services/firebaseService.ts
import auth from '@react-native-firebase/auth';
import firestore from '@react-native-firebase/firestore';
import analytics from '@react-native-firebase/analytics';
import crashlytics from '@react-native-firebase/crashlytics';

class FirebaseService {
  static getAuth() {
    return auth();
  }

  static getFirestore() {
    return firestore();
  }

  static getAnalytics() {
    return analytics();
  }

  static getCrashlytics() {
    return crashlytics();
  }

  // Initialize crash reporting
  static async initializeCrashlytics(): Promise<void> {
    // Crashlytics is automatically initialized in React Native Firebase
    // Additional setup can be done here if needed
  }

  // Log analytics event
  static async logEvent(name: string, parameters?: Record<string, any>): Promise<void> {
    try {
      await analytics().logEvent(name, parameters);
    } catch (error) {
      console.error('Analytics log error:', error);
    }
  }

  // Set user properties
  static async setUserProperty(name: string, value: string): Promise<void> {
    try {
      await analytics().setUserProperty(name, value);
    } catch (error) {
      console.error('Analytics user property error:', error);
    }
  }
}

export default FirebaseService;

